### ISSUE TITLE
Create tests / warning around high risk changes

### ISSUE THREAD

**Issue by davidkelliott** [ministryofjustice/modernisation-platform#5947]
### User Story

As an MP engineer
I want to be alerted to or prevented from making changes that may break the platform
So that I don't break the platform.

### Value / Purpose

There are certain changes we know should not be made, eg changing a json file name for an application, an others which are high risk, eg making any changes to the application json file.

By preventing or warning on the PR for these changes we reduce the risk of them accidentally getting past PR review and going through.

### Proposal A proposal that is something testable, don't worry whether it works or not, it's a place for ideas.
Add to the OPA tests changes that we don't want to be made.
Add a check which will warn if changes are high risk by posting a comment on the PR, along with any additional information which may help a reviewer make a decision.
Going forward we should then add additional tests and warning every time we have an incident which could have been prevented.

### Definition of Done

- [x] OPA tests updated
- [x] New warning tests which post to the PR created
- [x] Documentation has been written / updated
- [x] README has been updated
- [x] User docs have been updated
- [x] Another team member has reviewed
- [x] Tests are green

Accounts 

- [ ] Core-Shared-Services
- [ ] Core-Network-Services 
- [ ] Core-Logging 
- [ ] Core-Security
- [ ] Core-VPC-* 
- [ ] Modernisation-Platform
- [ ] PagerDuty
- [ ] Github

**Comment by dms1981** [comment]
We do have defensive tests in place at present using terratest - https://github.com/ministryofjustice/modernisation-platform/blob/main/.github/workflows/core-logging-deployment.yml#L71

Is this story meant to supplant these old tests and perform them through OPA?

**Comment by richgreen-moj** [comment]
So far I've raised this PR in draft https://github.com/ministryofjustice/modernisation-platform/pull/6281

The plan is to do the following

1. Turn on status checks for the opa policy testing workflow so that PRs cannot be merged without the tests passing.
2. Add a workflow/script that adds comments to PRs warning of the risks associated with making changes to various areas oof the code. 

I've also scheduled a meeting with the wider team to discuss any further test use cases we can implement.
